libbeat/processors/dissect/const.go

```diff
@@ -0,0 +1,32 @@
+package dissect
+
+import (
+	"errors"
+	"regexp"
+)
+
+var (
+	// delimiterRE tokenizes the following string into walkable with extracted delimiter + key.
+	// string:
+	// ` %{key}, %{key/2}`
+	// into:
+	// [["", "key" ], [", ", "key/2"]]
+	delimiterRE = regexp.MustCompile("(?s)(.*?)%\\{([^}]*?)}")
+	suffixRE    = regexp.MustCompile("(.+?)(/(\\d{1,2}))?(->)?$")
+
+	skipFieldPrefix      = "?"
```

Comment:
I wonder if we really need to support all of these prefixes or if we can leave the more complex ones to LS?

Reply:
Also the indirect is really useful, take this example:

`%{?key} %{&key}` -> "hello world" -> {"hello" => "world"}